ocaml/typing/typedtree.ml

```diff
@@ -1035,13 +1035,20 @@ let rev_let_bound_idents_full bindings =
   List.iter (fun vb -> iter_bound_idents add vb.vb_pat) bindings;
   !idents_full
 
-let let_bound_idents_with_modes_and_sorts bindings =
+let let_bound_idents_with_modes_sorts_and_checks bindings =
   let modes_and_sorts = Ident.Tbl.create 3 in
   let f id sloc _ _uid mode sort =
-    Ident.Tbl.add modes_and_sorts id (sloc.loc, mode, sort)
+    Ident.Tbl.add modes_and_sorts id
+      (sloc.loc, mode, sort, Builtin_attributes.Default_check)
   in
   List.iter (fun vb ->
-    iter_pattern_full ~both_sides_of_or:true f vb.vb_sort vb.vb_pat)
+    iter_pattern_full ~both_sides_of_or:true f vb.vb_sort vb.vb_pat;
+     match vb.vb_pat.pat_desc, vb.vb_expr.exp_desc with
+     | Tpat_var (id, _, _, _), Texp_function fn ->
+       let (loc, mode, sort, _) = Ident.Tbl.find modes_and_sorts id in
+       Ident.Tbl.replace modes_and_sorts id (loc, mode, sort, fn.zero_alloc)
```

Comment:
This `replace` makes me nervous. It modifies the most recently `add`-ed binding, and there may be multiple bindings added by `iter_pattern_full` (as evidenced by the `Ident.Tbl.find_all` below). When can there be multiple entries for the same ident?

Reply:
Thanks! Should `let_bound_idents_with_modes_sorts_and_checks` return a single check (rather than a list of checks)? That should simplify the client code, and it's easy enough to validate the assertion that "the list is either all Default_check or a singleton of something else" just by looking at this function.

Reply:
I agree, and have made this change.  I had plans of supporting more complicated patterns at some point, but it's totally unnecessary and anyway we can make this more complicated then if it ever happens.